lib/components/base/Pagination.vue

```diff
@@ -49,66 +49,63 @@
     </a>
   </div>
 </template>
-<script setup>
+<script setup lang="ts">
```

Comment:
On second thought, a separate PR for doing typescript conversions separate from the composition API conversions would be nice.

Reply:
I think split on components makes a lot more sense because the PRs will atomic and independent, thus can be accepted quicker (as long as you are going to review and merge).

Split on Composition and TypeScript conversions makes less sense, since what TypeScript conversion achieves the same goal that Composition conversion does. E.g., `defineProps` with TypeScript generic emits the same code that `defineProps` with options argument.

It's just twice the effort. There's no crazy TypeScript shenanigans going on to really justify a separate review. But it's up to you.